sub/internal/storage/platform/db/tx.go

```diff
@@ -0,0 +1,44 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/jmoiron/sqlx"
+
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-hrvadl/sub/internal/storage/transaction"
+)
+
+func NewWithTx(db *sqlx.DB) *Tx {
+	return &Tx{
+		db: db,
+	}
+}
+
+type Tx struct {
+	db *sqlx.DB
+}
+
+func (d *Tx) ExecContext(ctx context.Context, query string, args ...any) (sql.Result, error) {
+	tx, err := transaction.FromContext(ctx)
+	if err != nil {
```

Comment:
В принципі також робочий підхід, закину ще один варіант, може тобі сподобається.
Що якщо в `FromContext` повертати лише `*sqlx.Tx`, тоді цю перевірку можна буде замінити на:
```suggestion
	if tx == nil {
```
Семантично мені цей варіант більше подобається

Reply:
теоретично можна, проте я всюди в своєму коді роблю перевірки саме на err != nil, тому у цілях консистентності коду, може залишити?

Reply:
так, без проблем